// wdio-native-app-compare-service/build/helpers/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.NO_PLATFORM_VERSION = exports.NO_PLATFORM_NAME = exports.NO_ORIENTATION = exports.NO_DEVICE_NAME = exports.NO_APPEARANCE = exports.IPHONE_SMALL_SIZE = exports.IPHONE_MEDIUM_SIZE = exports.IPHONE_LARGE_SIZE = exports.IPHONE_EXTRA_LARGE_SIZE = exports.IOS_RECTANGLES = exports.DIFF = exports.DEFAULT_SCREENSHOTS_FOLDER = exports.DEFAULT_RESIZE_DIMENSIONS = exports.DEFAULT_IMAGE_FORMAT_NAME = exports.DEFAULT_BASELINE_FOLDER = exports.ACTUAL = void 0;
const ACTUAL = 'actual';
exports.ACTUAL = ACTUAL;
const DEFAULT_IMAGE_FORMAT_NAME = '{tag}-{deviceName}';
exports.DEFAULT_IMAGE_FORMAT_NAME = DEFAULT_IMAGE_FORMAT_NAME;
const DEFAULT_BASELINE_FOLDER = './baseline';
exports.DEFAULT_BASELINE_FOLDER = DEFAULT_BASELINE_FOLDER;
const DEFAULT_RESIZE_DIMENSIONS = {
  top: 0,
  right: 0,
  bottom: 0,
  left: 0
};
exports.DEFAULT_RESIZE_DIMENSIONS = DEFAULT_RESIZE_DIMENSIONS;
const DEFAULT_SCREENSHOTS_FOLDER = './.tmp/screenshots';
exports.DEFAULT_SCREENSHOTS_FOLDER = DEFAULT_SCREENSHOTS_FOLDER;
const DIFF = 'diff';
exports.DIFF = DIFF;
const NO_DEVICE_NAME = 'no-device-name-in-caps';
exports.NO_DEVICE_NAME = NO_DEVICE_NAME;
const NO_ORIENTATION = 'no-orientation-in-caps';
exports.NO_ORIENTATION = NO_ORIENTATION;
const NO_PLATFORM_NAME = 'no-platform-name-in-caps';
exports.NO_PLATFORM_NAME = NO_PLATFORM_NAME;
const NO_PLATFORM_VERSION = 'no-platform-version-in-caps';
exports.NO_PLATFORM_VERSION = NO_PLATFORM_VERSION;
const NO_APPEARANCE = 'no-appearance-in-caps'; //=====================
// iPhone X-series data
//=====================
// Height of screen in portrait mode, this is also what you get back
// when you:
//  - ask for the screen dimensions in WebdriverIO `getWindowSize`
//  - with `screen.height` in Safari devtools
//  - with http://whatismyscreenresolution.net/
// 6/7/8:                    667
// 6/7/8/Plus:               736
// X|XS|11Pro|12Mini|13Mini  812
// 12|12Pro|13|13Pro:        844
// XSMax|XR|11|11ProMax:     896
// 12ProMax|13ProMax         926

exports.NO_APPEARANCE = NO_APPEARANCE;
const IPHONE_SMALL_SIZE = 812;
exports.IPHONE_SMALL_SIZE = IPHONE_SMALL_SIZE;
const IPHONE_MEDIUM_SIZE = 844;
exports.IPHONE_MEDIUM_SIZE = IPHONE_MEDIUM_SIZE;
const IPHONE_LARGE_SIZE = 896;
exports.IPHONE_LARGE_SIZE = IPHONE_LARGE_SIZE;
const IPHONE_EXTRA_LARGE_SIZE = 926;
exports.IPHONE_EXTRA_LARGE_SIZE = IPHONE_EXTRA_LARGE_SIZE;
const IOS_RECTANGLES = {
  DEFAULT: {
    STATUS_BAR: 20,
    HOME_BAR: {
      PORTRAIT: {
        height: 0,
        width: 0,
        x: 0,
        y: 0
      }
    }
  },
  SMALL: {
    STATUS_BAR: 44,
    HOME_BAR: {
      PORTRAIT: {
        bottom: 6,
        right: 135,
        left: 120,
        top: 799
      }
    }
  },
  LARGE: {
    STATUS_BAR: 44,
    HOME_BAR: {
      PORTRAIT: {
        bottom: 6,
        right: 148,
        left: 133,
        top: 883
      }
    }
  },
  MEDIUM: {
    STATUS_BAR: 44,
    HOME_BAR: {
      PORTRAIT: {
        bottom: 7,
        right: 140,
        left: 125,
        top: 830
      }
    }
  },
  EXTRA_LARGE: {
    STATUS_BAR: 44,
    HOME_BAR: {
      PORTRAIT: {
        bottom: 7,
        right: 155,
        left: 136,
        top: 912
      }
    }
  }
};
exports.IOS_RECTANGLES = IOS_RECTANGLES;
